src/iuse_actor.cpp

```diff
@@ -2583,6 +2599,16 @@ repair_item_actor::attempt_hint repair_item_actor::repair( player &pl, item &too
     if( action == RT_REPAIR ) {
         if( roll == SUCCESS ) {
             pl.add_msg_if_player(m_good, _("You repair your %s!"), fix.tname().c_str());
+            if( t->has_quality( quality_sew ) ) {
+                p->add_msg_if_player(m_info,
+                    _("You have used %d filament charges."), filament_required);
+                for (const auto &it : req.get_components()) {
+                    p->consume_items(it);
+                }
+                for (const auto &it : req.get_tools()) {
+                    p->consume_tools(it);
+                }
```

Comment:
It is necessary to do something right now when sewing tools have lost their charges, otherwise textile repair won't spend any thread.

Game asks for filament only on SUCCESS roll, not on every tick - that is exactly the same like `handle_components` function is being called.